front/src/common/Map/WarpedMap/core/quadtree.ts

```diff
@@ -0,0 +1,112 @@
+/* eslint-disable prefer-destructuring, no-plusplus */
+import { BBox2d } from '@turf/helpers/dist/js/lib/geojson';
+import { Feature, FeatureCollection, GeoJsonProperties, Geometry, Position } from 'geojson';
+import bbox from '@turf/bbox';
+
+// The following types help describing a full QuadTree:
+export type Leaf<T> = {
+  type: 'leaf';
+  elements: T[];
+};
+export type Quad<T> = {
+  type: 'quad';
+  bbox: BBox2d;
+  children: [QuadChild<T> | null, QuadChild<T> | null, QuadChild<T> | null, QuadChild<T> | null];
+};
+export type QuadChild<T> = Quad<T> | Leaf<T>;
+
+export function bboxIntersect([mx1, my1, Mx1, My1]: BBox2d, [mx2, my2, Mx2, My2]: BBox2d): boolean {
+  return !(mx1 > Mx2) && !(Mx1 < mx2) && !(my1 > My2) && !(My1 < my2);
+}
+
+export function getNewQuadChild<T>(box: BBox2d, isLeaf?: boolean): QuadChild<T> {
```

Comment:
No need to export this function
```suggestion
function getNewQuadChild<T>(box: BBox2d, isLeaf?: boolean): QuadChild<T> {
```